### PR TITLE
NFTMethod.getNextAvailableIndex returns the highest index incremented by 1 within a collection

### LINKED ISSUE #8729: NFTMethod.getNextAvailableIndex iterates over all chains
### Expected behavior
NFTMethod.getNextAvailableIndex should only iterate over collections in `ownChain` and return the index by incrementing the highest index within the collection by 1.

Implement the suggested change in [LIP PR](https://github.com/LiskHQ/lips/pull/412).

### Actual behavior
NFTMethod.getNextAvailableIndex iterates over all chains.

### Which version(s) does this affect? (Environment, OS, etc...)
`feature/6917-implement-nft-module` branch.

### PR DESCRIPTION
**Pull request by has5aan**
### What was the problem?

This PR resolves #8729 

### How was it solved?

Updated `NFTMethod.getNextAvailableIndex` to return the highest index incremented by 1 within a collection.

### How was it tested?

Added/Updated unit tests.

### CODE REVIEW

File: framework/src/modules/nft/method.ts [hunk 1]
```diff
@@ -278,21 +278,29 @@ export class NFTMethod extends BaseMethod {
...
+		const indexBytes = latestKey.slice(LENGTH_CHAIN_ID + LENGTH_COLLECTION_ID, LENGTH_NFT_ID);
+		const index = indexBytes.readBigUInt64BE();
+		const largestIndex = BigInt(BigInt(2 ** 64) - BigInt(1));
+
+		if (index === largestIndex) {
+			throw new Error('No more available indexes');
 		}
 
-		return count;
+		return indexBytes.readBigUInt64BE() + BigInt(1);
```
Comment (mjerkov) [inline framework/src/modules/nft/method.ts:303]: I believe you can make this change:
```suggestion
		return index + BigInt(1);
```

**Review by mjerkov — APPROVED** [review]
All good, just a minor suggestion

File: framework/src/modules/nft/method.ts [hunk 2]
```diff
@@ -278,21 +278,29 @@ export class NFTMethod extends BaseMethod {
...
+		const indexBytes = latestKey.slice(LENGTH_CHAIN_ID + LENGTH_COLLECTION_ID, LENGTH_NFT_ID);
+		const index = indexBytes.readBigUInt64BE();
+		const largestIndex = BigInt(BigInt(2 ** 64) - BigInt(1));
+
+		if (index === largestIndex) {
+			throw new Error('No more available indexes');
 		}
 
-		return count;
+		return index + BigInt(1);
```
Comment (Incede) [inline framework/src/modules/nft/method.ts:303]: Apparently the [LIP PR](https://github.com/LiskHQ/lips/pull/412) was updated after this change. I think it needs to be updated to align per the [latest commit](https://github.com/LiskHQ/lips/pull/412/commits/944a1e71c4280f0001f1b84e7e3c58ba86f990eb)

**Review by Incede — CHANGES_REQUESTED** [review]

File: framework/src/modules/nft/method.ts (hunk 2 above)
Comment (mjerkov) [reply framework/src/modules/nft/method.ts:303]: I believe that now the LIP is closer to implementation than before, @has5aan please comment

Comment (mjerkov) [reply framework/src/modules/nft/method.ts:303]: On second thought: yes, it's closer regarding the store iteration, but the last part of the pseudocode has slightly changed

Comment (Incede) [reply framework/src/modules/nft/method.ts:303]: Looking more closely, I now see that the overall logic is still the same after the [commit](https://github.com/LiskHQ/lips/pull/412/commits/944a1e71c4280f0001f1b84e7e3c58ba86f990eb) but was only a syntactical change. I resolve my comment.

**Review by Incede — APPROVED** [review]

**Pull request merged**